src/components/App/App.js

```diff
@@ -1,22 +1,51 @@
-import React, { useState, useEffect } from 'react';
+import React, { useEffect, useReducer } from 'react';
 import './App.css';
 import { Start } from '../Start';
 import { RoundSelection } from '../RoundSelection';
 import { Round } from '../Round';
 import { Result } from '../Result';
 import { FinalResult } from '../FinalResult';
-import { CleanUp } from '../CleanUp/CleanUp';
 import { fetchAccessToken } from '../../api/fetchAccessToken';
 import { isAccessTokenValid } from '../../utils/isAccessTokenValid';
 import { saveAccessToken } from '../../utils/saveAccessToken';
+import { ACTIONS } from '../hooks/actions';
 
 const App = () => {
-    // todo: refactor to look nicer, maybe in a useReducer
-    const [route, setRoute] = useState('start');
-    const [rounds, setRounds] = useState(5);
-    const [albums, setAlbums] = useState([]);
-    const [currentRound, setCurrentRound] = useState(1);
-    const [results, setResults] = useState([]);
+    const reducer = (state, action) => {
+        switch (action.type) {
+            case ACTIONS.SET_ROUTE:
+                if (action.payload === 'reset') return { ...initialState };
```

Comment:
Small thing, but this could be a different action all together :) `ACTIONS.RESET`. It's not only setting the route to it's initial state, but also all the other state. I think it would make more sense for it to live in it's own action instead of being a special route reset action

Reply:
Ah that's right. I made it to mirror the old `useState` and`<Cleanup />` logic, but it should be in a separate action.